src/handler/mod.rs

```diff
@@ -0,0 +1,145 @@
+/// This module most of low-level commands
+/// and grub variable modifications
+use anyhow::{bail, Error, Result};
+use std::fs::OpenOptions;
+use std::io::Write;
+use std::process::Command;
+use std::str;
+
+/// reboots the system if boot_counter is greater 0 or can be forced too
+pub fn handle_reboot(force: bool) -> Result<(), Error> {
```

Comment:
There are several places where you are using `Result<(), Error>` and then in other places simply `Result<()>`.
When using `anyhow`, `Result<(), Error>` ( with`anyhow::Error`) is equivalent to `Result<()>` so we should use the later one since is shorter.

Reply:
I'm unresolving this conversation since I can still see `Result<(), Error>` elsewhere in the code.

Reply:
@say-paul please look at this too